rpc/core/blocks.go

```diff
@@ -83,19 +83,20 @@ func filterMinMax(base, height, min, max, limit int64) (int64, int64, error) {
 // Block gets block at a given height.
 // If no height is provided, it will fetch the latest block.
 // More: https://docs.tendermint.com/master/rpc/#/Info/block
-func Block(ctx *rpctypes.Context, heightPtr *int64) (*ctypes.ResultBlock, error) {
-	height, err := getHeight(env.BlockStore.Height(), heightPtr)
-	if err != nil {
-		return nil, err
-	}
-
-	block := env.BlockStore.LoadBlock(height)
-	blockMeta := env.BlockStore.LoadBlockMeta(height)
-	if blockMeta == nil {
-		return &ctypes.ResultBlock{BlockID: types.BlockID{}, Block: block}, nil
-	}
-	return &ctypes.ResultBlock{BlockID: blockMeta.BlockID, Block: block}, nil
-}
+// todo: use IPLD plugin interface
+// func Block(ctx *rpctypes.Context, heightPtr *int64) (*ctypes.ResultBlock, error) {
+// 	height, err := getHeight(env.BlockStore.Height(), heightPtr)
+// 	if err != nil {
+// 		return nil, err
+// 	}
+
+// 	block := env.BlockStore.LoadBlock(height)
+// 	blockMeta := env.BlockStore.LoadBlockMeta(height)
+// 	if blockMeta == nil {
+// 		return &ctypes.ResultBlock{BlockID: types.BlockID{}, Block: block}, nil
+// 	}
+// 	return &ctypes.ResultBlock{BlockID: blockMeta.BlockID, Block: block}, nil
+// }
```

Comment:
This would be a very simple function, that a) gets the block hash for the requested height, b) gets the DA header for that hash, c) uses the DA header to actually get the block data (e.g. via `RetrieveBlockData` of adr002 / the PRs that @evan-forbes is working on). Not sure if a) + b) could be one step instead.